src/dune_rules/melange/melange_binary.ml

```diff
@@ -1,7 +1,13 @@
 open Import
 
 let melc sctx ~loc ~dir =
-  Super_context.resolve_program_memo sctx ~loc ~dir ~hint:"opam install melange" "melc"
+  Super_context.resolve_program_memo
+    sctx
+    ~loc
+    ~dir
+    ~where:Original_path
+    ~hint:"opam install melange"
+    "melc"
```

Comment:
@anmonteiro I tweaked the resolution of melange here as well. It shouldn't affect anything, but could in theory help with some odd cycles